02-04-domain-validations/internal/platform/server/handler/courses/search_all.go

```diff
@@ -0,0 +1,38 @@
+package courses
+
+import (
+	"net/http"
+
+	mooc "github.com/CodelyTV/go-hexagonal_http_api-course/02-04-domain-validations/internal"
+	"github.com/gin-gonic/gin"
+)
+
+type CourseResponse struct {
+	Id       string
+	Name     string
+	Duration string
+}
+
+// CreateHandler returns an HTTP handler for courses creation.
+func SearchAllHandler(courseRepository mooc.CourseRepository) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+
+		courses, err := courseRepository.SearchAll(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, err.Error())
```

Comment:
Again, it's not directly related with the course's contents itself, but it looks weird to return a `400 Bad Request` where there are no requirements for the given request. I'd return a `500 Internal Server Error` (or a more specific one), considering what could fail here is the connection or the database itself. Thanks!

Reply:
Yes, I have to choose the status code returned better